replacy/__init__.py

```diff
@@ -136,28 +143,48 @@ def get_predicates(self, match_hooks) -> List[Callable]:
             predicates.append(pred)
         return predicates
 
-    def inflect_suggestion(self, pre_suggestion, doc, start, end, match_name):
+    def process_suggestions(self, pre_suggestion, doc, start, end, match_name):
         """
+        Perform inflection and replace references to the matched token
         example of pattern and pre_suggestion
         pattern: "LEMMA": "chock", "TEMPLATE_ID": 1
         pre_suggestion: "TEXT": "chalk", "FROM_TEMPLATE_ID": 1
         inflect suggestion "chalk" according to form of "cholk" from patterns
         """
         text_list = []
         for item in pre_suggestion:
-            text = item["TEXT"]
+            try:
+                text = item["TEXT"]
+            except KeyError:
+                ref = int(item["PATTERN_REF"])
+                refd_token = doc[start + ref]
```

Comment:
After this point we have `refd_token` (spacy token) OR `text` (string). Can we have one object with the same type? (See next comment. Maybe
```
text = refd_token.text_with_ws
```

Reply:
agreed this is sub-par.